hook.php

```diff
@@ -33,12 +33,14 @@
  * Entry point for installation process
  */
 function plugin_flyvemdm_install() {
-   global $DB;
-
+   $version   = plugin_version_flyvemdm();
+   $migration = new Migration($version['version']);
    require_once(PLUGIN_FLYVEMDM_ROOT . "/install/installer.class.php");
    $installer = new PluginFlyvemdmInstaller();
-
-   return $installer->install();
+   if (PluginFlyvemdmInstaller::getCurrentVersion() == '') {
+      return $installer->install($migration);
+   } 
+   return $install->upgrade($migration);
 }
```

Comment:
do not delegate to the method install() the task to distinguish a fresh installation and an upgrade